include/aws/common/math.h

```diff
@@ -91,10 +91,13 @@ AWS_STATIC_IMPL size_t aws_mul_size_saturating(size_t a, size_t b) {
 }
 
 AWS_STATIC_IMPL int aws_mul_size_checked(size_t a, size_t b, size_t *r) {
-    if ((uint64_t)SIZE_MAX == ~(uint32_t)0) {
-        return (int)aws_mul_u32_checked((uint32_t)a, (uint32_t)b, (uint32_t *)r);
-    }
-    return (int)aws_mul_u64_checked((uint32_t)a, (uint32_t)b, (uint64_t *)r);
+#if SIZE_MAX == UINT32_MAX
+    return aws_mul_u32_checked(a, b, (uint32_t *)r);
```

Comment:
These casts should no longer be required? If we can safely remove them, having 1 fewer cast could help the compiler find issues like this in the future.

Reply:
Weirdly, it seems to require the cast: if I remove it, I get
```
make[2]: *** [CMakeFiles/aws-c-common.dir/source/byte_buf.c.o] Error 1
In file included from /Users/dsn/ws/CCommon-CBMC/aws-c-common/source/date_time.c:17:
In file included from /Users/dsn/ws/CCommon-CBMC/aws-c-common/include/aws/common/array_list.h:19:
/Users/dsn/ws/CCommon-CBMC/aws-c-common/include/aws/common/math.h:97:38: error: incompatible pointer types passing 'size_t *' (aka 'unsigned long *') to parameter of type 'uint64_t *' (aka 'unsigned long long *') [-Werror,-Wincompatible-pointer-types]
    return aws_mul_u64_checked(a, b, r);
                                     ^
/Users/dsn/ws/CCommon-CBMC/aws-c-common/include/aws/common/math.gcc_overflow.inl:39:75: note: passing argument to parameter 'r' here
AWS_STATIC_IMPL int aws_mul_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
                                                                          ^
1 error generated.
make[2]: *** [CMakeFiles/aws-c-common.dir/source/date_time.c.o] Error 1
```